tools/sigclient/pkg/query/performance.go

```diff
@@ -0,0 +1,423 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package query
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/rand"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"verifier/pkg/utils"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type Query string
+
+const (
+	ComplexSearchQuery Query = "complexsearch"
+	StatsQuery               = "stats"
+	GroupByQuery             = "groupby"
+)
+
+var Queries = []Query{
+	ComplexSearchQuery,
+	StatsQuery,
+	GroupByQuery,
+}
+
+var pattern = `[^a-zA-Z0-9]`
+
+var globalQid = int64(0)
+
+// Compile the regular expression
+var patternRegex = regexp.MustCompile(pattern)
+
+const WAIT_DURATION_FOR_LOGS = 15 * time.Second
+
+var colsToIgnore = map[string]struct{}{
+	"_index":                       struct{}{},
+	"account":                      struct{}{},
+	"account_status":               struct{}{},
+	"account.balance":              struct{}{},
+	"account.created_data.country": struct{}{},
+	"account.created_data.date":    struct{}{},
+	"account.currency.long":        struct{}{},
+	"account.currency.short":       struct{}{},
+	"account.number":               struct{}{},
+	"account.type":                 struct{}{},
+	"timestamp":                    struct{}{},
+}
+
+var lowCardCols = map[string]int{
+	"gender":      2,
+	"bool_col":    2,
+	"http_method": 6,
+}
+
+var commonLock = &sync.RWMutex{}
+
+func GetQid() int64 {
+	return atomic.AddInt64(&globalQid, 1)
+}
+
+// Main function that tests all the queries
+func PerformanceTest(ctx context.Context, logChan chan utils.Log, dest string, concurrentQueries int, variableColNames []string) {
+
+	if ctx == nil {
+		log.Fatalf("PerformanceTest: ctx or logChan is nil")
+	}
+	if logChan == nil {
+		log.Fatalf("PerformanceTest: logChan is nil")
+	}
+
+	commonLock.Lock()
+	for _, col := range variableColNames {
+		colsToIgnore[col] = struct{}{}
+	}
+	commonLock.Unlock()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			wg := sync.WaitGroup{}
+			for i := 0; i < concurrentQueries; i++ {
+				wg.Add(1)
+				go func() {
+					RunPerfQueries(ctx, logChan, dest)
+					wg.Done()
+				}()
+			}
+			wg.Wait()
+			time.Sleep(time.Duration(rand.Intn(100)+50) * time.Millisecond)
+		}
+	}
+}
+
+func RunPerfQueries(ctx context.Context, logChan chan utils.Log, dest string) {
+
+	// Run all the queries _, query := range Queries
+	for _, query := range Queries {
+		var err error
+		select {
+		case logReceived := <-logChan:
+			if wait := logReceived.Timestamp.Add(WAIT_DURATION_FOR_LOGS).Sub(time.Now()); wait > 0 {
+				log.Warnf("Waiting for %v", wait)
+				select {
+				case <-time.After(wait):
+				case <-ctx.Done():
+					return
+				}
+			}
+			qid := GetQid()
+			switch query {
```

Comment:
nit: better name for this would be queryType

Reply:
addressed in the latest commit.